src/Configuration/Config.AzureKeyVault/src/AzureKeyVaultConfigurationSource.cs

```diff
@@ -8,7 +8,7 @@ namespace Microsoft.Extensions.Configuration.AzureKeyVault
     /// <summary>
     /// Represents Azure KeyVault secrets as an <see cref="IConfigurationSource"/>.
     /// </summary>
-    internal class AzureKeyVaultConfigurationSource : IConfigurationSource
```

Comment:
I'm sure the product team will scream about this.  Let's change this back to internal and get rid of the other constructor.

Reply:
We would expose this object's methods and members to other namespaces. Not sure why they wanted it that way, but this would be a big deal.

Reply:
In other extensions (such as JSON and file) this class is public to allow the user more fine-tuned control over the configuration of the provider.
Unless we want to expose every single possible configuration source member as a parameter, which would make the code more cumbersome to use, then this is the only option.